likecoin/public/likecoin-button.php

```diff
@@ -42,73 +45,46 @@ function likecoin_add_likebutton( $likecoin_id = '' ) {
 		}
 	}
 
-	if ( strlen( $likecoin_id ) > 0 ) {
-		$referrer     = is_preview() ? '' : '&referrer=' . rawurlencode( get_permalink( $post ) );
-		$sandbox_attr = function_exists( 'is_amp_endpoint' ) && is_amp_endpoint() ? 'sandbox="allow-scripts allow-same-origin allow-popups allow-popups-to-escape-sandbox allow-top-navigation-by-user-activation allow-storage-access-by-user-activation" ' : '';
-		$widget_code  = '<figure class="likecoin-embed likecoin-button"><iframe scrolling="no" frameborder="0" ' . $sandbox_attr .
-		'style="height: 212px; width: 100%;" ' .
-		'src="https://button.like.co/in/embed/' . $likecoin_id . '/button' .
-		'?type=wp' . $referrer . '"></iframe></figure>';
-		return $widget_code;
-	}
-	return '';
-}
-
-/**
- * Add LikeButton to post content if suitable
- *
- * @param string| $content The original post content.
- */
-function likecoin_content_filter( $content ) {
-	global $post;
-	$option          = get_option( LC_BUTTON_OPTION_NAME );
-	$post_type_query = '';
-
-	if ( ! isset( $post ) ) {
-		return $content;
-	}
 	do {
 		// follow post meta if option is not set yet.
 		if ( ! isset( $option[ LC_OPTION_BUTTON_DISPLAY_AUTHOR_OVERRIDE ] ) || $option[ LC_OPTION_BUTTON_DISPLAY_AUTHOR_OVERRIDE ] ) {
 			$widget_option   = get_post_meta( $post->ID, LC_OPTION_WIDGET_OPTION, true );
 			$widget_position = isset( $widget_option[ LC_OPTION_WIDGET_POSITION ] ) ? $widget_option[ LC_OPTION_WIDGET_POSITION ] : '';
+			if ( isset( $option[ LC_OPTION_BUTTON_DISPLAY_OPTION ] ) ) {
+				$type = $option[ LC_OPTION_BUTTON_DISPLAY_OPTION ];
+				switch ( $type ) {
+					case 'none':
+						$post_type_query = 'none';
```

Comment:
While using `none` would very likely to work, `is_singular( 'none' )` would actually query custom post type "none". Would we set a flag when $type is 'none', then do not render button accordingly?